scripts/builtin/mice_nominal.dml

```diff
@@ -0,0 +1,72 @@
+#-------------------------------------------------------------
+#
+# Copyright 2019 Graz University of Technology
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+#-------------------------------------------------------------
+
+
+# Builtin function Implements Multiple Imputation using Chained Equations (MICE) for nominal data
+#
+# INPUT PARAMETERS:
+# ---------------------------------------------------------------------------------------------
+# NAME            TYPE    DEFAULT     MEANING
+# ---------------------------------------------------------------------------------------------
+# F               String    ---        Data Frame
+# iter            Integer    3         Number of iteration for multiple imputations 
+# complete        Integer    3         A complete dataset generated though a specific iteration
+# ---------------------------------------------------------------------------------------------
+ 
+
+#Output(s)
+# ---------------------------------------------------------------------------------------------
+# NAME                  TYPE    DEFAULT     MEANING
+# ---------------------------------------------------------------------------------------------
+# dataset               Double   ---        imputed dataset
+# singleSet             Double   ---        A complete dataset generated though a specific iteration
+
+# Assumption missing value are represented with empty string i.e ",," in csv file  
+
+s_mice_nominal= function(Frame[String] F, Integer iter = 3, Integer complete = 3)
+  return(Frame[String] dataset, Frame[String] singleSet)
+{
+  if(ncol(F) == 1)
+    stop("invalid aregument: can not apply mice on single column")
+  # detecting the schema of data to identify nominal columns
+  schema = detectSchema(F);
+  s=""
+  for(i in 1: ncol(schema))
+  {
+    if(as.scalar(schema[1, i]) == 'STRING' | as.scalar(schema[1, i]) == 'BOOLEAN' )
+      s = s+i+","
+  }
+  # encoding nominal columns using recode transformation
+  jspec = "{ids:true, recode:["+s+"]}";
+  [X, M] = transformencode(target=F, spec=jspec);
```

Comment:
The transform currently has no support for creating code that preserves the order of nominal features - I think we would need to add this first. Otherwise, the subsequent lm calls will give you "random" results.

Reply:
Maybe a simpler start would be to support categoricals (including nominals) and simply call a classification algorithm instead of lm